storage/teststore/queue.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package teststore
+
+import (
+	"container/list"
+	"fmt"
+
+	"storj.io/storj/storage"
+)
+
+//Queue is a non-threadsafe FIFO queue implementing DistQueue
+type Queue struct {
+	*list.List
```

Comment:
`container/list` is rarely the right answer. You can just use `[]storage.Value`, we can optimize later when needed.

Also it should be threadsafe, wrapping it in a mutex should be sufficient.

Reply:
This is a test/mock class.  Do we really care about its implementation?  If the goal is to create something that valid as a distributed queue, does a mutex actually help?

Reply:
We will have tests that probably end up using it from multiple goroutines.

Reply:
added mutex